hydra-node/test/Hydra/Chain/Direct/WalletSpec.hs

```diff
@@ -227,6 +232,37 @@ isBalanced utxo originalTx balancedTx =
 ledgerPParams :: PParams (ShelleyLedgerEra Era)
 ledgerPParams = toLedgerPParams (shelleyBasedEra @Era) Fixture.pparams
 
+prop_picksCorrectUTxOToPayTheFees :: Property
+prop_picksCorrectUTxOToPayTheFees =
+  forAllBlind genUTxO $ \lookupUTxO ->
+    forAllBlind genMarkedUTxO $ \fuelUTxO ->
+      case findFuelOrLargestUTxO (Map.union lookupUTxO fuelUTxO) of
+        Nothing ->
+          property False
+            & counterexample ("Lookup UTXO: \n" <> decodeUtf8 (encodePretty lookupUTxO))
+            & counterexample ("Fuel UTXO: \n" <> decodeUtf8 (encodePretty fuelUTxO))
+        Just (txin, txout) -> property $ Map.singleton txin txout == fuelUTxO
+
+prop_picksLargestUTxOToPayTheFees :: Property
+prop_picksLargestUTxOToPayTheFees =
+  forAllBlind genUTxO $ \utxo1 ->
+    forAllBlind genUTxO $ \utxo2 -> do
+      let combinedUtxO = Map.union utxo1 utxo2
+      case findFuelOrLargestUTxO combinedUtxO of
+        Nothing ->
+          property False
+            & counterexample ("Sample UTXO: \n" <> decodeUtf8 (encodePretty combinedUtxO))
+        Just (txin, txout) ->
+          property $
+            largestUTxO == expectedLargestUTxO
+              & counterexample ("Sample UTXO: \n" <> decodeUtf8 (encodePretty combinedUtxO))
+              & counterexample ("Max found UTXO: \n" <> decodeUtf8 (encodePretty largestUTxO))
+              & counterexample ("findFuelOrLargestUTxO found UTXO: \n" <> decodeUtf8 (encodePretty expectedLargestUTxO))
+         where
+          fromLedgerUTxO = bimap Api.fromLedgerTxIn Api.fromLedgerTxOut
+          largestUTxO = UTxO.maxLovelaceUTxO $ UTxO.fromPairs $ fromLedgerUTxO <$> Map.toList combinedUtxO
+          expectedLargestUTxO = UTxO.singleton $ fromLedgerUTxO (txin, txout)
```

Comment:
Should: not re-use the function which finds the max utxo here, instead we should rely on a property we know must hold after selection. For example: `txOutValue txout >= utxo1 && txOutValue txout >= utxo2`